lib/src/model/lobby/game_seek.dart

```diff
@@ -113,4 +117,70 @@ sealed class GameSeek with _$GameSeek {
     if (variant != null) 'variant': variant!.name,
     if (ratingRange != null) 'ratingRange': '${ratingRange!.$1}-${ratingRange!.$2}',
   };
+
+  factory GameSeek.fromJson(Map<String, dynamic> json) => _$GameSeekFromJson(json);
+}
+
+class RecentGameSeekPrefs implements Serializable {
+  final List<GameSeek> seeks;
+
+  const RecentGameSeekPrefs({required this.seeks});
+
+  @override
+  Map<String, dynamic> toJson() {
+    return {'seeks': seeks.map((e) => e.toJson()).toList()};
+  }
+
+  factory RecentGameSeekPrefs.fromJson(Map<String, dynamic> json) {
+    final rawList = json['seeks'] as List<dynamic>? ?? [];
+    return RecentGameSeekPrefs(
+      seeks: rawList.map((item) => GameSeek.fromJson(item as Map<String, dynamic>)).toList(),
+    );
+  }
+
+  RecentGameSeekPrefs copyWithAdded(GameSeek newRequest) {
+    final existing = seeks.where((r) => r != newRequest).toList();
+    final updated = [newRequest, ...existing];
+    if (updated.length > 3) {
+      updated.removeRange(3, updated.length);
+    }
+    return RecentGameSeekPrefs(seeks: updated);
+  }
+
+  static const empty = RecentGameSeekPrefs(seeks: []);
+}
+
+final recentGameSeekProvider = NotifierProvider<RecentGameSeekNotifier, RecentGameSeekPrefs>(
```

Comment:
I don't really mind but for consistency with the rest you could use the `@riverpod` notation.
I also don't like much the generator; hopefully it will be removed with a next version of riverpod.